App/components/DatabaseComponent.js

```diff
@@ -0,0 +1,180 @@
+import React from "react";
+import {
+  ScrollView,
+  StyleSheet,
+  Text,
+  View,
+  TouchableOpacity,
+} from "react-native";
+import Constants from "expo-constants";
+import { FlatList, ForceTouchGestureHandler, TouchableHighlight } from "react-native-gesture-handler";
+import * as SQLite from "expo-sqlite";
+import CustomButton from "../../App code/CustomButton";
+import { Button } from "react-native";
+const db = SQLite.openDatabase("grocery_Items.db");
+
+class Items extends React.Component 
+{
+  constructor() {
+  super();
+  this.state = {
+    items: null,
+    ColorHolder: "#252525",
+  }};
+
+  componentDidMount() {
+    this.update();
+  }
+
+ markItemOwned=()=>{
+    if(this.state.ColorHolder != "coral"){
+      
+      this.setState({
+        ColorHolder: "coral"
+      })
+    }
+    else{
+      this.setState({
+        ColorHolder: "#252525"
+      })
+    }
+  }
+
+  render() {
+    const { items } = this.state;
+
+    if (items === null || items.length === 0) {
+      return null;
+
+    }
+
+    return (
+      <View style={styles.sectionContainer}>
+        {/* Map loop to iterate through the database and show them in a Text component */}
+        {items.map(({ id, name, expirationDate, price }) => (
+          <TouchableOpacity key={id} style={styles.itemcontainer,{backgroundColor: this.state.ColorHolder}} onPress={this.markItemOwned}>

```

Comment:
So this is where you mark the item if you have it right? How does it know what state it is in?

Reply:
State wise these item buttons are always assigned the ColorHolder value with is #252525 and upon pressing the button the state of ColorHolder value is changed to color coral which therefore changes the color of item. Short to the point I am using dynamic stylesheets to manipulate the state to adjust the value.